clients/appointments/src/components/index.tsx

```diff
@@ -82,26 +84,33 @@ export const AppointmentsView = ({
   }
 
   const onCancel = () => {
-    putAppointment({
-      onComplete: afterCancel,
-      onError: handleError,
-      setLoading,
-      appointmentCoding: {
-        code: getAppointmentType(appointmentCancellation.appointment.code),
-      },
-      locationId: appointmentCancellation.appointment.locationId || locationId,
-      timeSlot: {
-        start: appointmentCancellation.appointment.start,
-        end: appointmentCancellation.appointment.end,
-        provider: {
-          id: appointmentCancellation.appointment.providerId,
+    try {
+      putAppointment({
+        onComplete: afterCancel,
+        onError: handleError,
+        setLoading,
+        appointmentCoding: {
+          code:
+            appointmentCancellation?.appointment?.code ||
+            defaultAppointmentType.code,
```

Comment:
This seems like the actual fix, to ensure it sends a valid code?

Reply:
The actual fix was to remove the `getAppointmentType` call in our code. This is to give a reasonable default if for some reason we can't get the `code` from the appointment. 

Jack requested this FWIW